### PR TITLE
VSCODE-86: Playgrounds and connections explorers update

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

- Add playgrounds explorer to the tree views
- Search for `.mongodb` files in the current workspace including all nested folders and excluding folders and files specified in MongoDB extension settings
- User can manage the list of excluded folders and files
- Support the refresh action in the playgrounds explorer contextual menu
- Support the create a new playground action in the playgrounds explorer contextual menu
- Open the playground by clicking on it or using a contextual menu
- For connections use the connections explorer header instead of the connections tree root

Kapture 2020-08-25 at 17 24 28 4

Screenshot 2020-08-25 at 17 11 56

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by Anemy** [comment]
Hmm what should happen if a file was deleted, and the playgrounds list hasn't been refreshed yet, and the now non-existant playground is clicked on?
I'm thinking maybe we can show a prompt or something, currently it silently fails and passes an error to console:
Screen Shot 2020-08-27 at 10 28 34 AM

**Comment by alenakhineika** [comment]
@Anemy I tested `mdb.showMongoDBConnectionExplorer` and `showMongoDBPlaygrounds` manually and they don't call getChildren if set to false. I didn't find a good way hot to cover it with unit tests, so we can add them later. Fell free to add them later as well, if you will have some ideas.

Updated PR to address other comments, thx for the valuable feedback!

File: package.json [hunk 1]
```diff
@@ -300,6 +323,14 @@
         "command": "mdb.refreshIndexes",
         "title": "Refresh"
       },
+      {
+        "command": "mdb.createPlaygroundFromTreeView",
```
[the remark this answers is not in the document]
Comment (Anemy) [reply package.json:327]: Just this comment - then lgtm!

[the remark this answers is not in the document]
Comment (alenakhineika) [reply package.json]: Uh, i thought i deleted it! Now deleted for sure :)

File: package.json [hunk 2]
```diff
@@ -168,10 +179,18 @@
         "command": "mdb.createPlayground",
         "title": "MongoDB: Create MongoDB Playground"
       },
+      {
+        "command": "mdb.refreshPlaygrounds",
```
Comment (Anemy) [inline package.json:183]: Ah I think this one makes it into the command palette.
```
...
"editor/title": [{
  "command": "mdb. refreshPlaygrounds",
  "when": "false"
}
...]
```
Is it something we want in the command palette? In that case maybe we could add a command for that which has the `MongoDB:` prefix

**Review by Anemy — APPROVED** [review]
One small command tweak that's in the command palette, otherwise lgtm!
Nice work on this.

File: package.json (hunk 2 above)
Comment (alenakhineika) [reply package.json:183]: Added additional command for refreshing playgrounds from the command palette in addition to one is being called from the tree view. Plus test for it.

**Review by Anemy — APPROVED** [review]
lgtm!

**Pull request merged**